### ISSUE TITLE
the 'date' param of 'onChange' is not a javascript Date object

### ISSUE THREAD

**Issue by YaoHuiJi** [farhoudshapouran/react-native-ui-datepicker#91]
Would returning a standard JavaScript Date type instead of dayjs object be more appropriate? Otherwise, it could be unintuitive and inconvenient for users who don't use the dayjs library(they have to use `param.date.toDate()` which is not mentioned in the document)

And I also found that when set `timePicker=true`, `the param.date` of onChange is a string? These behaviors are inconsistent, unintuitive and inconvenient. Is it designed this way for some reason?

**Comment by thecodecafe** [comment]
@YaoHuiJi I had the same thought, I want to use this library but the only thing turning me away is the dependency on DaysJS I think the standed Date object that comes with javascript should be the date object used.

**Comment by GustavoBonfimS** [comment]
It would be interesting if the onChange output was a javascript Date object, or at least sent as another parameter.


Can the project maintainers discuss this? It would be great to hear their opinion on this.

**Comment by GustavoBonfimS** [comment]
It seems to be possible to use the Date object with this little workaround

```tsx
function DatePickerModal({
	modal: { getParam },
}: ModalProps<"DatePickerModal">) {
  const defaultDate = getParam("date");
  const [selectedDate, setSelectedDate] = useState(defaultDate ?? new Date());

  return (
     <DefaultContainer variant="modal">
	<DateTimePicker
            onChange={({ date }) => {
              setSelectedDate(new Date(date as string));
            }}
           date={selectedDate}
           mode="single"
           locale="pt-br"
         />

       <Button>Confirmar</Button>
    </DefaultContainer>
    );
}

export default DatePickerModal;
```

**Comment by demmyhonore** [comment]
The _dayjs_ dependency (2kb) does make it a breeze to apply localisation to the date picker. 

But yeah, to keep things "open" it would be nice to have Date objects in the change handlers.

**Comment by ronKovler** [comment]
couldnt you just do `onChange={(params: {date: DateType}) => setDate(params.date as Date)}`? Works for me

**Comment by thecodecafe** [comment]
@ronKovler that won't be a good idea as DateType and Date are not the same object, you would instead need to fully convert DateType to a Date by using.

**Comment by MoOx** [comment]
Using TypeScript, a safe way to get a Date object is to do the following

```ts
onChange={({ date }) => {
  if (date && typeof date === "object" && "toDate" in date) {
    const d = date.toDate();
    console.log("change", d);
  }
}}
```

This should be in doc for sure.

**Comment by farhoudshapouran** [comment]
Yes, you are right it returns the dayjs object I will change it to the javascript Date object in the next release. in some days 👀

**Comment by farhoudshapouran** [comment]
Hi again after v3.0.3 all onChange methods `single`, `range`, and `multiple` return javascript Date object.

**Comment by farhoudshapouran** [comment]
> [@YaoHuiJi](https://github.com/YaoHuiJi) I had the same thought, I want to use this library but the only thing turning me away is the dependency on DaysJS. The standard Date object in Javascript should be the date object used.

after v3 you don't need to install dayjs anymore if you don't need it.

**Comment by Grohden** [comment]
@farhoudshapouran `onChange` will still return all possible types?

https://github.com/farhoudshapouran/react-native-ui-datepicker/blob/f3b5d1b2a88472098063cda7a9095d1b41e86e0b/src/types.ts#L6

**Comment by TheAlexDev23** [comment]
@farhoudshapouran Any updates on this? Is it safe to assume the onChange parameter will be of type `Date`
